woocommerce/woocommerce-ios
# Integrates wc-analytics/leaderboards API

Closes https://github.com/woocommerce/woocommerce-ios/issues/2422

# Why

This PR takes over the work on adapting the new `wc-analytics/leaderboards` API into the top performers' view of the app.
This is mainly done in order to solve some inconsistency problems between the TopEarners API and the web dashboard.
 
# How

The `wc-analytics/leaderboards` is not mobile-friendly and misses some of the data. This [issue](woocommerce/woocommerce#32274) asks wc-admin team to improve it.

While that is improved, we have decided to take the approach of converting the `leaderboards` into our current `TopEearnersStats` object, the rationale is to not rearchitect our model if/when this API improves.

 In order to do ☝️  we have to:
- Extract the productIDs from the `leaderboards` API, which come inside an HTML.
- Fetch those products from the internal storage or the remote service.
- Assemble the `TopEearnersStats` object with the leaderboard + product information

To accomplish that, this PR adds changes to 3 main components:
-  `LeaderboardsRemote`: Requests leaderboard API and parses its response them into the concrete `Leaderboard` data structure.
- `LeaderboardStatsConverter`: Provides a set of static helpers to allow us to convert a `Leaderboard` into a `TopEarnerStat` object.
- `StatsStoreV4`: Adds changes to orchestrate `LeaderboardsRemote` and `LeaderboardStatsConverter` when the app requests top performers data.

**PS: This PR has gotten way too big and I couldn't split it successfully without losing the commit history :-(**
**Let me know if you still prefer it split**

# Screenshots

Web|App
----|----
<img width="577" alt="web" src="https://user-images.githubusercontent.com/562080/88244864-bf7d3c00-cc5a-11ea-95b0-000a25a859be.png">|<img width="421" alt="app" src="https://user-images.githubusercontent.com/562080/88244866-c015d280-cc5a-11ea-8fe7-25024c8ac0fe.png">


# Testing Steps
- Launch the app and navigate through the top performer's section
- Make sure the values displayed in the app are the same as in the web WC dashboard for the given time period.

Update release notes:
